File: mybatis-plus-join-test/test-kotlin/src/main/java/com/github/yulichang/test/kotlin/entity/AddressDO.java
package com.github.yulichang.test.kotlin.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.experimental.Accessors;

@Data
@ToString
@Accessors(chain = true)
@EqualsAndHashCode
@TableName("address")
public class AddressDO {

    @TableId
    private Integer id;

    @TableField("user_id")
    private String userID;

    private String tel;

    private String address;
}

File: mybatis-plus-join-test/test-kotlin/src/main/kotlin/com/github/yulichang/test/kt/entity/AddressDO.kt
package com.github.yulichang.test.kt.entity

import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableName
import java.io.Serializable

@Suppress("unused")
@TableName("address")
class AddressDO : AddressGeneric<Int?, Int?, Int?, String?, String?, Boolean?>(), Serializable {
    @TableField(exist = false)
     var aaa = false
}
